api/routes/techieotm.py

```diff
@@ -0,0 +1,92 @@
+from datetime import datetime
+from fastapi import APIRouter, Depends, HTTPException, Query, status
+from sqlalchemy import desc, extract
+from sqlalchemy.orm import Session
+from api.api_models.user import TechieOTMCreate, TechieOTMPaginated, TechieOTMResponse
+from db.database import get_db
+from utils.permissions import is_admin
+from db.models.users import TechieOTM, User
+
+
+techieotm_router = APIRouter(tags=["User"], prefix="/users/techieotm")
+
+
+@techieotm_router.post("/", status_code=status.HTTP_201_CREATED, response_model=TechieOTMResponse)
+def create_techie_of_the_month(techieotm: TechieOTMCreate, current_user=Depends(is_admin), db: Session = Depends(get_db)):
+    current_month =datetime.now().month
+    current_year = datetime.now().year
+
+    existing_techieotm = db.query(TechieOTM).filter(extract("month", TechieOTM.created_at) == current_month, extract("year", TechieOTM.created_at) == current_year).first()
+    if existing_techieotm:
+        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail="Techie of the Month already posted for the current month")
+
+    user = db.query(User).filter(User.id == techieotm.user_id).first()
+    if not user:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="User not found")
+
+    new_techieotm = TechieOTM(**techieotm.dict())
+
+    db.add(new_techieotm)
+    db.commit()
+    db.refresh(new_techieotm)
+
+    techieotm_response = TechieOTMResponse(
+        id=new_techieotm.id,
+        user=user,
+        points=new_techieotm.points,
+        created_at=new_techieotm.created_at,
+    )
+
+    return techieotm_response
+
+
+@techieotm_router.get("/latest", response_model=TechieOTMResponse)
+def get_latest_techie_of_the_month(db: Session = Depends(get_db)):
```

Comment:
Why is this an entire endpoint? Can't we pass `limit` and `order` to the same endpoint for getting all?

Reply:
well we can, but this is what came to mind